lib/modules/manager/bazel-module/parser.ts

```diff
@@ -0,0 +1,71 @@
+import { lang, query as q } from 'good-enough-parser';
+import { logger } from '../../../logger';
+import { regEx } from '../../../util/regex';
+import { Ctx } from './context';
+import type { ValueFragment } from './fragments';
+import { StarlarkBoolean } from './starlark';
+
+const booleanValuesRegex = regEx(`^${StarlarkBoolean.stringValues.join('|')}$`);
+const supportedRules = ['bazel_dep'];
+const supportedRulesRegex = regEx(`^${supportedRules.join('|')}$`);
+
+/**
+ * Matches key-value pairs:
+ * - `name = "foobar"`
+ * - `dev_dependeny = True`
+ **/
+const kwParams = q
+  .sym<Ctx>((ctx, token) => {
+    return Ctx.as(ctx).startAttribute(token.value);
+  })
+  .op('=')
+  .alt(
+    q.str((ctx, token) => {
+      return Ctx.as(ctx).addString(token.value);
+    }),
+    q.sym<Ctx>(booleanValuesRegex, (ctx, token) => {
+      return Ctx.as(ctx).addBoolean(token.value);
+    })
+  );
+
+const moduleRules = q
+  .sym<Ctx>(supportedRulesRegex, (ctx, token) => {
+    return Ctx.as(ctx).startRule(token.value);
+  })
+  .join(
+    q.tree({
+      type: 'wrapped-tree',
+      maxDepth: 1,
+      search: kwParams,
+      postHandler: (ctx, tree) => {
+        return Ctx.as(ctx).endRule();
+      },
+    })
+  );
+
+const rule = q.alt<Ctx>(moduleRules);
+
+const query = q.tree<Ctx>({
+  type: 'root-tree',
+  maxDepth: 16,
+  search: rule,
+});
```

Comment:
We're using the same name `q` for different things, and the name `q` is not very descriptive.

Can we use unique and human-readable names instead?

Reply:
Sorry. I don't understand. There is a single `q` defined as part of the import. I followed the pattern used in the `gradle`, `graddle-wrapper`, `bazel`, `sbt`, and `pip_setup` managers.

Reply:
Oh if we use that pattern in more places then the maintainers are probably okay with it. Ignore my comment. 🙂